internal/services/mssql/helper/database.go

```diff
@@ -96,13 +99,21 @@ func FindDatabaseReplicationPartners(ctx context.Context, databasesClient *sql.D
 					if err != nil {
 						return nil, fmt.Errorf("retrieving Partner %s: %+v", partnerDatabaseId, err)
 					}
+
 					if location.NormalizeNilable(partnerDatabase.Location) != location.Normalize(*linkProps.PartnerLocation) {
 						log.Printf("[INFO] Mismatch of possible Partner Database based on location (%s vs %s) for %s", location.NormalizeNilable(partnerDatabase.Location), location.Normalize(*linkProps.PartnerLocation), id)
 						continue
 					}
+
 					if partnerDatabase.ID != nil {
-						log.Printf("[INFO] Found Partner %s", partnerDatabaseId)
-						partnerDatabases = append(partnerDatabases, partnerDatabase)
+						// TODO: add check here to make sure the enclave type of the partnerDatabaseId matches the primaryEnclaveType
+						// Once the fix for 'Azure Bug 2805551 ReplicationLink API ListByDatabase missed subsubcriptionId
+						// in partnerDatabaseId in response body' has been deployed in Azure. For now adding a placeholder for
+						// where the check should be once fixed...
+						if string(primaryEnclaveType) != pointer.From(partnerDatabase.DatabaseProperties.SourceDatabaseID) {
```

Comment:
The `databaseClient` here is using the old legacy client (e.g., `databasesClient *sql.DatabasesClient`) and the `enclave_type` is not exposed in the `v5.0` API so this was my best effort just to call out that this will need to be fixed once the regression for the `ListByDatabaseComplete` is live.